assets/js/baseline-grid.js

```diff
@@ -0,0 +1,108 @@
+/* global Prince */
+
+// Lint with JS Standard
+
+function ebAlignToGrid (querySelectorString) {
+  // Get all elements we want aligned with baseline grid.
+  // Note that unlike InDesign we do not force elements
+  // to the next baseline unit. Rather, we add margin-bottom
+  // so that they end aligned. Then any following elements
+  // will also be aligned with the baseline grid.
+  // This is mainly because Prince collapses margin-top
+  // on elements that appear at the top of a page,
+  // and we can't add margin-top to those, nor distinguish them
+  // from other elements in order to use a different algorithm.
+  // We use margin, not padding, because in testing it has been
+  // more reliable (e.g. padding on a table doesn't add space).
+
+  let elementsToAlign = document.querySelectorAll('.content h1, .content h2, .content h3, .content h4, .content h5, .content math[display="block"], .content p, .content ol, .content ul, .content table, .content .figure, .content .video, .content .box, .content .align-to-baseline')
+
+  const elementsToLeave = document.querySelectorAll('.content li > p, .content .figure *')
+
+  if (querySelectorString) {
+    elementsToAlign = document.querySelectorAll(querySelectorString)
```

Comment:
@arthurattwell did you mean to redefine `elementsToAlign` here? Or did you mean instead to append this new nodelist to the existing list of elements to align? As it stands, it looks as though setting a value for `querySelectorString` will result in only those selected elements being aligned.

Reply:
@LouiseSteward Thanks for checking. I did mean to redefine here. The idea is that it should be possible to run `alignToGrid('.boopbeep')` separately, and have it only align `boopbeep` elements. In a forthcoming commit, I'll also make it possible to pass the elements to leave/not-align in this way.